pkg/forward/metrics.go

```diff
@@ -28,77 +28,77 @@ var readMessagesCount = promauto.NewCounterVec(prometheus.CounterOpts{
 	Subsystem: "forwarder",
 	Name:      "read_total",
 	Help:      "Total number of Messages Read",
-}, []string{metrics.LabelVertex, metrics.LabelPipeline, "buffer"})
+}, []string{metrics.LabelVertex, metrics.LabelPipeline, "partition_idx"})
```

Comment:
I am wondering if we care about the metric per in-coming edge. With this dimension definition, we can't tell how much data is read for a specific in-coming edge.

Reply:
since the edge is owned by the vertex now, it's the same as how much data is read in the vertex.

Reply:
Sorry let me re-phase my comment, when we have diamond shape pipeline, we will have multiple edges owned by the same vertex. In that case, with the current metric definition, we won't be able to tell the read_total per edge. I guess for this PR, diamond shape is out of scope.

Reply:
we will still be reading from the same set of partitions.. diamond DAG is out of scope for this pr.